prac_07/programming_language.py

```diff
@@ -0,0 +1,44 @@
+"""
+CP1404/CP5632 Practical - Suggested Solution
+Programming Language class with tests.
+"""
+
+
+class ProgrammingLanguage:
+    """Represent information about a programming language."""
+
+    def __init__(self, name, typing, reflection, pointer_arithmetic, year):
+        """Construct a ProgrammingLanguage from the given values."""
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.pointer_arithmetic = pointer_arithmetic
+        self.year = year
+
+    def __repr__(self):
+        """Return string representation of a ProgrammingLanguage."""
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, Pointer Arithmetic=" \
+               f"{self.pointer_arithmetic}, First appeared in {self.year}"
+
+    def is_dynamic(self):
+        """Determine if language is dynamically typed."""
+        return self.typing == "Dynamic"
+
+
+def run_tests():
+    """Run simple tests/demos on ProgrammingLanguage class."""
```

Comment:
Good job remembering to add the pointer arithmetic field to the run_test function, but I think you may have misread whether ruby and python have pointer arithmetic